File: src/test/java/com/tuuzed/feedparser/FeedParserTest.java
package com.tuuzed.feedparser;

import org.junit.Test;

import java.io.FileReader;
import java.net.URL;


public class FeedParserTest {

    @Test
    public void rss() throws Exception {
        FeedParser.parse("http://epayment.blog.sohu.com/rss", new FeedCallbackImpl());
        FeedParser.parse("http://paynews.net/portal.php?mod=rss", new FeedCallbackImpl());
        FeedParser.parse("http://news.qq.com/newsgn/rss_newsgn.xml", new FeedCallbackImpl());
        FeedParser.parse("http://feedmaker.kindle4rss.com/feeds/choice.thepaper.xml", new FeedCallbackImpl());
    }

    @Test
    public void redirect_rss() throws Exception {
        FeedParser.parse("http://www.zhihu.com/rss", new FeedCallbackImpl());
    }

    @Test
    public void redirect_atom() throws Exception {
        FeedParser.parse("http://www.v2ex.com/feed/tab/tech.xml", new FeedCallbackImpl());
    }

    @Test
    public void atom() throws Exception {
        FeedParser.parse("https://www.v2ex.com/feed/tab/tech.xml", new FeedCallbackImpl());
    }

    @Test
    public void localRss() throws Exception {
        URL url = FeedParserTest.class.getClassLoader().getResource("rss20.xml");
        if (url != null) {
            FeedParser.parse(new FileReader(url.getFile()), new FeedCallbackImpl());
        }
    }


    @Test
    public void localAtom() throws Exception {
        URL url = FeedParserTest.class.getClassLoader().getResource("atom10.xml");
        if (url != null) {
            FeedParser.parse(new FileReader(url.getFile()), new FeedCallbackImpl());
        }
    }

}

File: src/test/kotlin/com/tuuzed/feedparser/FeedParserTest.kt
package com.tuuzed.feedparser

import org.junit.Test
import java.io.FileReader


class FeedParserTest {

    @Test
    @Throws(Exception::class)
    fun rss() {
        FeedParser.parse("http://epayment.blog.sohu.com/rss", FeedCallbackImpl())
        FeedParser.parse("http://paynews.net/portal.php?mod=rss", FeedCallbackImpl())
        FeedParser.parse("http://news.qq.com/newsgn/rss_newsgn.xml", FeedCallbackImpl())
    }

    @Test
    @Throws(Exception::class)
    fun atom() {
        FeedParser.parse("https://www.v2ex.com/feed/tab/tech.xml", FeedCallbackImpl())
    }


    @Test
    @Throws(Exception::class)
    fun localRss() {
        val url = FeedParserTest::class.java.classLoader.getResource("rss20.xml")
        FeedParser.parse(FileReader(url!!.file), FeedCallbackImpl())
    }

    @Test
    @Throws(Exception::class)
    fun localAtom() {
        val url = FeedParserTest::class.java.classLoader.getResource("atom10.xml")
        FeedParser.parse(FileReader(url!!.file), FeedCallbackImpl())
    }

}